typescript/sdk/src/hook/HyperlaneHookDeployer.ts

```diff
@@ -0,0 +1,150 @@
+import debug from 'debug';
+
+import {
+  OptimismISM,
+  OptimismISM__factory,
+  OptimismMessageHook,
+  OptimismMessageHook__factory,
+  TestRecipient,
+  TestRecipient__factory,
+} from '@hyperlane-xyz/core';
+import { types } from '@hyperlane-xyz/utils';
+
+import { HyperlaneContracts, HyperlaneContractsMap } from '../contracts';
+import { HyperlaneDeployer } from '../deploy/HyperlaneDeployer';
+import { MultiProvider } from '../providers/MultiProvider';
+import { ChainMap, ChainName } from '../types';
+
+import { HookFactories, hookFactories } from './contracts';
+import { HookConfig, NativeType } from './types';
+
+export class HyperlaneHookDeployer extends HyperlaneDeployer<
+  HookConfig,
+  HookFactories
+> {
+  constructor(multiProvider: MultiProvider) {
+    super(multiProvider, hookFactories, {
+      logger: debug('hyperlane:HookDeployer'),
+    });
+  }
+
+  async deploy(
+    configMap: ChainMap<HookConfig>,
+  ): Promise<HyperlaneContractsMap<HookFactories>> {
+    const deployedContractMap = await super.deploy(configMap);
+    const hookAddress = deployedContractMap.test1.optimismMessageHook.address;
+
+    this.logger(`Setting hook address ${hookAddress} for OptimismISM`);
+    await deployedContractMap.test2.optimismISM.setOptimismHook(hookAddress);
+
+    return deployedContractMap;
+  }
+
+  async deployContracts(
+    chain: ChainName,
+    hookConfig: HookConfig,
+  ): Promise<HyperlaneContracts<HookFactories>> {
+    let optimismISM, optimismMessageHook, testRecipient;
+    if (hookConfig.nativeType === NativeType.ISM) {
+      optimismISM = await this.deployOptimismISM(
+        chain,
+        hookConfig.nativeBridge,
+      );
+      testRecipient = await this.deployTestRecipient(
+        chain,
+        optimismISM.address,
+      );
```

Comment:
this shouldn't be deployed here
fwiw we have these on the chains you are forking already